homeassistant/components/homekit/type_sensors.py

```diff
@@ -98,6 +101,114 @@ def update_state(self, entity_id=None, old_state=None, new_state=None):
                           self.entity_id, humidity)
 
 
+@TYPES.register('AirQualitySensor')
+class AirQualitySensor(HomeAccessory):
+    """Generate a AirQualitySensor accessory as air quality sensor."""
+
+    def __init__(self, hass, entity_id, name, **kwargs):
+        """Initialize a AirQualitySensor accessory object."""
+        super().__init__(name, entity_id, CATEGORY_SENSOR, **kwargs)
+
+        self.hass = hass
+        self.entity_id = entity_id
+
+        service = add_preload_service(self, SERV_AIR_QUALITY_SENSOR,
+                                      [CHAR_AIR_PARTICULATE_DENSITY])
+        self.char_quality = service.get_characteristic(CHAR_AIR_QUALITY)
+        self.char_quality.value = 0
+        self.char_density = service \
+            .get_characteristic(CHAR_AIR_PARTICULATE_DENSITY)
+        self.char_density.value = 0
+
+    def update_state(self, entity_id=None, old_state=None, new_state=None):
+        """Update accessory after state change."""
+        if new_state is None:
+            return
+
+        density = convert_to_float(new_state.state)
+        if density is not None:
+            self.char_density.set_value(density)
+            self.char_quality.set_value(AirQualitySensor.map_quality(density))
+            _LOGGER.debug('%s: Set to %d', self.entity_id, density)
+
+    @staticmethod
+    def map_quality(density):
+        """Map PM2.5 density to HomeKit AirQuality level."""
+        if density <= 35:
+            return 1
+        elif density >= 36 and density <= 75:
+            return 2
+        elif density >= 76 and density <= 115:
+            return 3
+        elif density >= 116 and density <= 150:
+            return 4
+        return 5
```

Comment:
Keep in mind that you should be able to handle float values from `convert_to_float`. What happens if `density = 35.1`?

You could do something like this (untested):
```py
from collections import OrderedDict
map = OrderedDict(sorted({1: 0, 2: 35, 3: 76, 4: 116, 5: 151}.items(), reverse=True))

def density_to_air_quality(density):
    for key, value in map.items():
        if density > value:
            return key
    return 0

Reply:
`density = 35.1`, It's my carelessness. Thank you for correcting it.
About `OrderedDict`, I keep the original `if` statements, since it's simple and has more theoretic performance in this simple case.

Reply:
Works as well.